docs/deploy-lab.md

```diff
@@ -0,0 +1,355 @@
+# Enabling blue/green deployment with canary
+
+Automating deployment is critical to the success of most projects. However, in today's world, it's not enough to just deploy your code. You also need to ensure downtime is limited (or eliminated), and you can quickly rollback in the event of a failure. One common approach to this is to use a [blue/green deployment strategy](https://martinfowler.com/bliki/BlueGreenDeployment.html). This strategy involves deploying your code to a new environment, and then slowly shifting traffic from the old environment to the new environment. This allows you to test your new code in production, and quickly rollback if there are any issues.
+
+In this tutorial, you'll explore how to use [Azure Container Apps](https://learn.microsoft.com/en-us/azure/container-apps) to deploy a new version of your application, and then use Azure Traffic Manager to shift traffic from the old environment to the new environment. You'll start by creating and configuring the necessary Azure resources. You'll then configure a workflow in GitHub to deploy your application with a [canary](https://www.opsmx.com/blog/what-is-canary-deployment/) to automate rollback. Finally, you'll test the deployment and rollback process by commiting a change and monitoring the workflow.
+
+## Create an Azure account and install resources
+
+This workshop uses Azure Container Apps as the cloud-based host for the NGINX application. You'll need to create an Azure account and install the Azure CLI to complete this tutorial.
+
+> **NOTE**: While this tutorial uses Azure Container Apps, the concepts and techniques can be applied to any cloud-based host.
+
+1. Create an [Azure account](https://azure.microsoft.com/free/) if you don't already have one
+2. Install the [Azure CLI](https://learn.microsoft.com/cli/azure/install-azure-cli)
+
+## Create the initial container app
+
+With the tooling installed, you can now create the container app. This will be the initial version of the application, and will be used as the baseline for the blue/green deployment. You will login to Azure using the Azure CLI, install the extension for Azure Container Apps, clone the starting repository, and then deploy the container app.
+
+1. Open a terminal or command window
+2. Run the following command to login to Azure for the Azure CLI:
+
+    ```bash
+    az login
+    ```
+
+3. Follow the prompts to login using a browser window
+4. Run the following command to install the `containerapp` extension:
+
+    ```bash
+    az extension add --name containerapp --upgrade
+    ```
+
+5. Run the following command to create a resource group for the container app:
+
+    ```bash
+    az group create --name my-container-app-rg --location westus
+    ```
+
+5. Clone your repository locally, replacing <YOUR_GITHUB_ACCOUNT_NAME> with your account name
+
+    ```bash
+    git clone https://github.com/<YOUR_GITHUB_ACCOUNT_NAME>/platform.git
```

Comment:
This seems to imply that the user has forked the repository - is that correct? Just asking because I don't see that noted above but seems like a good way to handle it 👍

Reply:
I should have asked before typing this part out. It was my understanding they would have forked the repository before reaching this point. I'm happy to add the steps